### PR TITLE
Feat: adding use-pinheader hook and test case

### LINKED ISSUE #312: Implement `usePinHeader` (accept pin labels as argument similar to chip)
See `useChip` for example

/bounty $5

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
-> Creating use-pinheader hook which accepts a string array of pinlabels as argument.
-> Creating a test case for it.

closes #312 
/claim #312 

cc: @seveibar

### CODE REVIEW

File: lib/hooks/use-pinheader.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import type { PinHeaderProps } from "@tscircuit/props"
+import { createUseComponent } from "./create-use-component"
+
+export const usePinHeader = (pinLabels: string[]) =>
+  createUseComponent(
+    (props: PinHeaderProps) => <pinheader pinLabels={pinLabels} {...props} />,
```
Comment (seveibar) [inline lib/hooks/use-pinheader.tsx:6]: Needs to have name prop as first argument, second argument should be { pinLabels } (an options style object). Use the pinlabelspec or PinHeaderProps[“pinLabels”] to accept all forms of pinlabels including object variants

**Review by seveibar — CHANGES_REQUESTED** [review]
Very close

File: lib/hooks/use-pinheader.tsx (hunk 1 above)
Comment (DhairyaMajmudar) [reply lib/hooks/use-pinheader.tsx]: Done the changes 👍🏻

File: lib/hooks/use-pinheader.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import type { PinHeaderProps } from "@tscircuit/props"
+import { createUseComponent } from "./create-use-component"
+
+export const usePinHeader = (
+  name: string,
+  options: {
+    pinLabels: PinHeaderProps["pinLabels"]
+  },
```
Comment (seveibar) [inline lib/hooks/use-pinheader.tsx:8]: Looking at your code vs other examples like useChip, i think “options” should actually be props: PinHeaderPrips

File: tests/hooks/use-pinheader.test.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const circuit = new Circuit()
+
+  const useHeader = usePinHeader("Header", {
+    pinLabels: ["VCC", "GND", "TX", "RX"],
+  })
+
+  const Header = useHeader("Header", {
+    pinCount: 4,
+    pinLabels: ["VCC", "GND", "TX", "RX"],
+  })
```
Comment (seveibar) [inline tests/hooks/use-pinheader.test.tsx:15]: The usage is wrong, see the useChip or useResistor examples.


const J1 = usePinHeader(“J1”, { pinLabels })

J1.myPin // string

<J1 />

**Review by seveibar — CHANGES_REQUESTED** [review]

File: tests/hooks/use-pinheader.test.tsx (hunk 3 above)
Comment (DhairyaMajmudar) [reply tests/hooks/use-pinheader.test.tsx:15]: gotcha, I've fixed the variable names 👍🏻

**Review by seveibar — CHANGES_REQUESTED** [review]
a hook shouldn't return another hook, only use the `use*` prefix for hooks

**Pull request closed without merging**